### PR TITLE
feat: Updated Huddle01 App Flow

### PR DESCRIPTION
**Pull request by ponikar**
## What does this PR do?

- We at Huddle01 have revamped several things, including how users interact and communicate through the Huddle01 app. Huddle01 now has a dedicated OAuth mechanism that empowers third-party applications like CalCom. To make CalCom compatible with the new flow, we have rewritten the current app.
- Introduced an Authentication page to allow users to confirm their identity before scheduling a call.
- Added Huddle01 OAuth support that works independently.
- Introduced a new set of APIs to communicate with Huddle01 servers.

https://www.loom.com/share/d3d725a07fc54ae0b7cb95a608b183d2?sid=b5c0a6fd-3766-45e8-83d4-c38134e3088f.

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Once you install a fresh Huddle01 app, you are required to confirm your identity via login to the Huddle01 auth page.
- Once the authentication process succeeds, you can configure your events to use the Huddle01 app to schedule meetings when users make bookings.
- Now, while booking a call, the user should see the Huddle01 app to book the call.
- API calls should work, and the user should see the meeting link after the booking is made.
- Test modify and cancel flows; we have added separate API calls corresponding to each functionality.
- On the booking pages, meeting details should be visible.

- **Are there environment variables that should be set?**
  - Yes, we have a new `HUDDLE01_API_TOKEN` to handle the authentication process. It needs to be set.

- **What are the minimal test data requirements?**
  - Huddle01 account to test authentication.

- **What is expected (happy path) to have (input and output)?**
  - Install the Huddle01 app and complete the authentication process.
  - Configure your events to use the Huddle01 app for scheduling meetings.
  - Users should be able to schedule, modify, and cancel the meeting.

## Checklist

- [ ] I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- [ ] My code doesn't follow the style guidelines of this project.
- [ ] I haven't commented my code, particularly in hard-to-understand areas.
- [x] I haven't checked if my changes generate no new warnings.

### CODE REVIEW

**Review by anikdhabal — CHANGES_REQUESTED** [review]
Hey @ponikar, can you please uncommit the lock file? I don't think it's need to update for the current changes.

File: packages/app-store/huddle01video/api/callback.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  const state = decodeOAuthState(req);
+
+  if (!session) {
+    return res.status(401).json({ message: "Unauthorized" });
+  }
+
+  const userId = session.user.id;
+
+  const { identityToken } = req.query;
```
Comment (anikdhabal) [inline packages/app-store/huddle01video/api/callback.ts:21]: grrg

File: packages/app-store/huddle01video/api/callback.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  const token = Array.isArray(identityToken) ? identityToken[0] : identityToken;
+  if (!token) {
+    return res.status(401).json({
+      message: "Something went wrong",
+    });
+  }
+
```
Comment (anikdhabal) [inline packages/app-store/huddle01video/api/callback.ts:29]: grg

**Comment by anikdhabal** [comment]
Thanks @ponikar , Could you also please add a README to explain the process of obtaining the Huddle API token? Like this:- https://github.com/calcom/cal.com/blob/main/packages/app-store/webex/README.md

**Comment by ponikar** [comment]
> Hey @ponikar, can you please uncommit the lock file? I don't think it's need to update for the current changes.

Hey @anikdhabal 
We don't provide third party authentication publicly as of now. We have internal APIs to create token to identify Calcom internally. It shouldn't affect calcom users to any other ways. 

If anyone requires to generate a similar token they can reach out to support@huddle01.com 

Feel free to share concern regarding any of it.

File: yarn.lock [hunk 3]
```diff
@@ -49876,4 +49876,4 @@ __metadata:
   resolution: "zwitch@npm:2.0.4"
   checksum: f22ec5fc2d5f02c423c93d35cdfa83573a3a3bd98c66b927c368ea4d0e7252a500df2a90a6b45522be536a96a73404393c958e945fdba95e6832c200791702b6
   languageName: node
-  linkType: hard
+  linkType: hard
```
Comment (anikdhabal) [inline yarn.lock:49879]: Remove unnecessary changes. Need one line space

File: .env.appStore.example [hunk 4]
```diff
@@ -137,3 +138,7 @@ REVERT_PUBLIC_TOKEN=
 # NOTE: If you're self hosting Revert, update this URL to point to your own instance.
 REVERT_API_URL=https://api.revert.dev/
 # *********************************************************************************************************
+
+#   - Huddle01
+# Used for the huddle01 integration
+HUDDLE01_API_TOKEN=
```
Comment (anikdhabal) [inline .env.appStore.example:144]: same here

**Comment by ponikar** [comment]
> Hey @ponikar I got you! Could you please provide a Loom video of the entire workflow?

Sure. Our team will make a video to understand the entire flow.

**Comment by anikdhabal** [comment]
> Sure. Our team will make a video to understand the entire flow.

I mean from our app's perspective: starting with installation from our app store using OAuth, then book an event with this app

File: .env.appStore.example (hunk 4 above)
Comment (anikdhabal) [reply .env.appStore.example:144]: I suggest that you leave one extra line. It currently ends at 144, line should end at 145. Just press Enter

**Comment by ponikar** [comment]
> > Sure. Our team will make a video to understand the entire flow.
> 
> I mean from our app's perspective: starting with installation from our app store using OAuth, then book an event with this app

Thanks for the detailed requirement.

**Comment by ponikar** [comment]
@anikdhabal we have some concern regarding how calcom will manage huddle01 flow for older users who is using the old version. Can we connect over slack or email? Please provide your email address to communicate further.

**Comment by ponikar** [comment]
> @anikdhabal we have some concern regarding how calcom will manage huddle01 flow for older users who is using the old version. Can we connect over slack or email? Please provide your email address to communicate further. 

Hey @anikdhabal any updates on this? We are creating a video but want to clarify this as well.

**Comment by ponikar** [comment]
@anikdhabal 
Here's video demonstration https://www.loom.com/share/d3d725a07fc54ae0b7cb95a608b183d2?sid=b5c0a6fd-3766-45e8-83d4-c38134e3088f. 

Let me know if you need anything else.

**Comment by anikdhabal** [comment]
@ponikar thanks for the loom video, I will give the final review.

**Comment by ponikar** [comment]
@anikdhabal Hey any updates on this?

**Comment by anikdhabal** [comment]
@ponikar Thanks for your work. But I have a suggestion for users who have already installed it

I think we can keep the same app and implement oauth for new install. But just making sure that legacy ones still works. And if someone wants to use the new one they can just uninstall and reinstall. We could also provide some banner for people with the legacy version

**Comment by ponikar** [comment]
Thanks for the response. I will talk with my team and will get back to you by end of the day.

**Comment by ponikar** [comment]
Hey @anikdhabal

Thanks for the response, We are good with the flow you suggested, We will support backward compatibly for old users. 

> We could also provide some banner for people with the legacy version
Please share more details on this, we would want to encourage old users to update the app. 

Also, If everything is good, can you please tell me where this PR will get merge and deploy?

**Comment by ponikar** [comment]
Hey @anikdhabal please let me know if we are good with merging this PR, let me know if I have to make any changes. 
Thanks

**Review by anikdhabal — APPROVED** [review]
Great Work @ponikar

**Comment by ponikar** [comment]
Thank you so much @anikdhabal.

**Comment by ponikar** [comment]
@anikdhabal We can see the Calcom app has been updated. However when we are trying to test new flow its scheduling meeting to Calcom video. Also your team didn't ask for `API_KEY` without this you can't communicate with our backend. 

Please let me know if I have missed something from my end. Users are not able to create meetings and they are reaching out to us.

**Comment by ponikar** [comment]
cc @AkMo3

**Comment by anikdhabal** [comment]
Hey @ponikar, have you tried uninstalling and reinstalling it? Does it fall back to the call video? If the problem persists with the new flow, then I think I will check that. Also, were you able to authenticate and install the app successfully? If possible, could you please share a video of the whole flow?

**Comment by ponikar** [comment]
Hi @anikdhabal 
Yes I reinstalled the app, the authentication worked and yes it fallbacks to call video. We were not able to see any logs from the backend to debug this issue.

**Comment by anikdhabal** [comment]
Thanks @ponikar will get back to this

**Comment by ponikar** [comment]
@anikdhabal We noticed that your team has disabled our old application without notifying us on prior. This is important for our users. Please restore the old application till the time your team figure out the fix. 

Please consider this one as priority.

**Comment by anikdhabal** [comment]
@ponikar what do you mean by disabling the old application? Could you pls explain? Btw it's there in the app store, maybe disabled and quickly enable it for testing

**Comment by ponikar** [comment]
calcom disabled our app and sent email to all our users. We have no idea why your team did this?

image

**Comment by ponikar** [comment]
I am requesting you to take this a priority as this is affecting out partners. Please inform us on priority. Is there any way to reach out you via slack? This is urgent and important we can afford much delay on this.

**Comment by anikdhabal** [comment]
> calcom disabled our app and sent email to all our users. We have no idea why your team did this?

Ah, this one! Don’t panic; I was just working to fix the current issue in the morning. I disabled it and then enabled it again. Now there’s no issue, and the app is working. Inform it other
Screenshot 2024-10-30 130449

**Comment by ponikar** [comment]
Thanks @anikdhabal if it is possible can we talk about this on slack/discord, this is really important for us. 
also what will happen now, will all users use new version? 

Our new version is not working as well, its falling back to cal video. We cannot debug this from our end.

**Comment by ponikar** [comment]
Can you send this confirmation email to all our existing users if possible, our users got panicked when they received this email.

**Comment by anikdhabal** [comment]
@ponikar let's talk about it in discord. Pls share you your link

**Comment by arcinston** [comment]
hi @anikdhabal , Arush here from Huddle01 Engineering team

this email has been sent to all our addon users which has lead to user panic and lots of support tickets opening

would request you to send an email as a followup informing all users that the addon is enabled

this would help swiftly solve this issue

**Comment by arcinston** [comment]
this is the link to our discord community : https://discord.gg/d9587UPR

**Comment by arcinston** [comment]
my username on discord : arcinston

send across a friend request and lets resolve this ASAP 
thanks

**Pull request merged**